web/pandas/pdeps/0008-inplace-methods-in-pandas.md

```diff
@@ -0,0 +1,402 @@
+fact that the ``inplace=True`` changes the return type of a method from a pandas object to ``None``, makes usage of
+the ``inplace`` keyword confusing and non-intuitive.
+
+In addition, some methods, such as ``DataFrame.rename`` and ``DataFrame.rename_axis`` confusingly support both
+the ``copy`` and ``inplace`` keywords, with the value of ``inplace`` overwriting the value of ``copy``.
+
+To summarize the status quo of inplace behavior of methods, we have divided methods that can operate inplace or have
+an ``inplace``/``copy`` keyword into 4 groups:
+
+**Group 1: Methods that always operate inplace**
+
+| Method Name   |
+|:--------------|
+| ``insert``    |
+| ``pop``       |
+| ``update``    |
+| ``isetitem``* |
+
+These methods always operate inplace and don't have the ``inplace`` or ``copy`` keyword.
+
+\* Although ``isetitem`` operates on the original pandas object inplace, it will not change any existing values
+inplace (it will remove the values of the column being set, and insert new values).
+
+**Group 2: Methods that modify the underlying data of the DataFrame/Series object and can be done inplace**
+
+| Method Name     |
+|:----------------|
+| ``where``       |
+| ``fillna``      |
+| ``replace``     |
+| ``mask``        |
+| ``interpolate`` |
+| ``ffill``       |
+| ``bfill``       |
+| ``clip``        |
+
+These methods don't operate inplace by default, but have the option to specify `inlace=True`. All those methods leave
+the structure of the DataFrame or Series intact (shape, row/column labels), but can mutate some elements of the data of
+the DataFrame or Series.
+
+**Group 3: Methods that modify the DataFrame/Series object, but not the pre-existing values**
+
+| Method Name                 |
+|:----------------------------|
+| ``drop`` (dropping columns) |
+| ``eval``                    |
+| ``rename``                  |
+| ``rename_axis``             |
+| ``reset_index``             |
+| ``set_index``               |
+| ``astype``                  |
+| ``infer_objects``           |
+| ``set_axis``                |
+| ``set_flags``               |
+| ``to_period``               |
+| ``to_timestamp``            |
+| ``tz_localize``             |
+| ``tz_convert``              |
+| ``swaplevel``               |
+| ``concat``                  |
+
+These methods can change the structure of the DataFrame or Series, such as changing the shape by adding or removing
+columns, or changing the row/column labels (changing the index/columns attributes), but don't modify the existing
+underlying data of the object.
+All those methods (except for `set_flags`) make a copy of the full data by default, but can be performed inplace with
+avoiding copying all data (currently enabled with the `inplace` or `copy` keyword).
+
+Some of these methods only have a `copy` keyword instead of an `inplace`
+keyword: `astype`, `infer_objects`, `set_axis`, `set_flags`, `to_period`, `to_timestamp`, `tz_localize`, `tz_convert`, `swaplevel`, `concat`
```

Comment:
Nit: Would be great to see this in the table, for example, having 2 extra columns llike "has copy", "has inplace"

Reply:
I added a keyword column for groups 3/4 (where there are multiple keywords).

I don't think the column is necessary for group 1 (no keyword) or group 2(only `inplace` keyword), but I can add it there as well for consistency and to make the info easier to find.